### ISSUE TITLE
Is there a way to add a 3rd middle point for blinds?

### ISSUE THREAD

**Issue by lacedup** [aarons22/homebridge-bond#158]
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

**Comment by aarons22** [comment]
Can you elaborate a little more? I don't have smart blinds myself so it's a little difficult to understand less straight forward requests like this. Would be great to see screenshots from the bond app and Home app (and some markup with what you expect).

**Comment by MattGeale** [comment]
> 

I think what they're wanting, as I just found this myself, is a way to stop the blinds at lets say 50% (ie 0% fully open, 100% being fully closed). I'd be wanting the same but unsure if Bond can even receive that information back from the blinds as most of them don't report their 'position status'...I think the only ones that I know that actively report their position are native homekit ones by Somfy, but the rest of their range doesn't; Lutron obviously being the exception, but they have their own native Homekit hub.

If this could be integrated in the form of lets say a timer to stop the blinds, that'd be wicked...I havent the foggiest how Bond gets the open/close status from the blinds (if it does at all) but I'm guessing there'd be some sort of broadcast from the blinds to report to the hub that says "the blind is closed, it cannot go up/down any further". If thats the case, maybe a timer could be worked into this to calculate how long it takes the blinds to go from 0 - 100 and if you want to stop at point X, just divide that time into however many segments or positions you want it to stop at. idk if thats even an option with Bond, but I'd be down to help out with testing if thats a feature that isn't in place. Cheers!

**Comment by chrismerck** [comment]
FWIW, on the Bond Bridge Pro, we support SetPosition for Somfy RTS shades. It works by deadreckoning with a Kalman filter and a little planning algorithm to avoid the pitfalls of the My and Stop signals being the same at the RF level. It works pretty well in practice.

For Rollease (and a few others), we use the native SetPosition RF signal.

At this time no shades provide RF feedback of position to the Bond Bridge, but that's something we're looking to add to the Bond Bridge Pro (which has better receive capabilities) when the shade supports it natively (Rollease, etc., not Somfy RTS sadly.)

tl;dr:  If you're using Somfy RTS and a Bond Bridge Pro, go to Device Settings > Advanced > Edit Controller, enable Position, and then select "Calibrate your Shade". You'll get a screen like below, and then the app will give a slider (and the API will give SetPosition/position feedback to HomeBridge).

<img width="375" alt="Screen Shot 2021-12-06 at 10 48 33" src="https://user-images.githubusercontent.com/628921/144877286-a0d054bc-453d-4059-a673-f41c2bacbe7b.png">

**Comment by MattGeale** [comment]
> 

Oh thats the answer I was wanting to hear regarding the SetPosition. I didnt know it worked like that! so theoretically we can set scenes or something to basically do "set shades at 50%" or something and they'll stop at 50% if calibrated correctly with this? You're making it very easy for me to pull the trigger on getting a Bond Bridge as this was one of the things that was stopping me from doing so just yet ;)

Cheers mate :)

**Comment by chrismerck** [comment]
Yes! Though to be clear the Somfy RTS position control is only available on the BD-1750-PRO model, not the BD-1000 model.

**Comment by MattGeale** [comment]
> Yes! Though to be clear the Somfy RTS position control is only available on the BD-1750-PRO model, not the BD-1000 model.

Dumb question- but how do you figure out what model shades you have? Unfortunately the company that I contracted to build my blinds and install the motors just put “Somfy RTS Motors” on the invoice. I haven’t called them yet, but just wondering if there is any discernible differences to identify.

I know that I have the “new models”; blind company words, not mine 🤷🏼‍♂️

**Comment by chrismerck** [comment]
> how do you figure out what model shades you have?

Indeed that can be a hard problem. Sometimes the motors don't even have model numbers printed on them.

But, what I was referring to was the model of the Bond Bridge. We have a Bond Bridge BD-1000 and a Bond Bridge Pro BD-1750-PRO. The Pro version has a bunch of features that go beyond the consumer version.

**Comment by MattGeale** [comment]
> > how do you figure out what model shades you have?
> 
> Indeed that can be a hard problem. Sometimes the motors don't even have model numbers printed on them.
> 
> But, what I was referring to was the model of the Bond Bridge. We have a Bond Bridge BD-1000 and a Bond Bridge Pro BD-1750-PRO. The Pro version has a bunch of features that go beyond the consumer version.

Oh lol - I’m dumb. I thought you were referencing the blinds themselves. I was thinking the Bond Bridge Pro anyway.

Thanks again!!

**Comment by MattGeale** [comment]
Hey @chrismerck,

These are my blind motors...are they compatible with the SetPosition?

[Sonnesse 2/20Nm Li-ion Motor](https://www.somfysystems.com/en-us/products/1003128/sonesse-30wirefree-rts-li-ion)

Cheers!

**Comment by MattGeale** [comment]
So do u have any list of compatible Somfy devices that set position works with? Think the terminology blinds/ shades have got me confused…I have roller blinds/shades that these motors are in.
Cheers

**Comment by chrismerck** [comment]
Any Somfy RTS shade/blind product that works with Open/Close/Stop type remote controls should work with SetPosition also.

**Comment by MattGeale** [comment]
This is the remote I use - sorry for all these questions, just want to make sure we’re in business before I pull the trigger on a bond bridge as this is one thing I really wanted to do with homebridge. 

[Somfy Telis 16 RTS] (https://www.somfy.com.au/products/1811020/telis-16-rts) 

thanks again for answering all my questions…

**Comment by chrismerck** [comment]
Yes, it should work.

**Comment by MattGeale** [comment]
@chrismerck Merry Christmas and Happy New Year mate! Just a bit of an update; I received my bridge pro and got my blinds setup and connected/configured using Set position. using the bond bridge, SetPosition works most times...its a little buggy (maybe?) that in the sense when you finish configuring them, the bridge shows the slider @ 50% and requires it to change for it to respond to setposition... I still havent figured out how to get SetPosition working on Siri so if I'm missing any sort of configuration or commands, let me know. Thanks again for your advise and all your help!

**Comment by chrismerck** [comment]
> SetPosition works most times...its a little buggy (maybe?) that in the sense when you finish configuring them, the bridge shows the slider @ 50% and requires it to change for it to respond to setposition... I

This is because we don't actually know the position of the shade. There is no feedback from the shade. There's also no signal for "SetPosition". We do deadreckoning and a Kalman filter to estimate when to send open/close/My signals. It's kind of amazing that it works at all. Basically, the shades don't support it, but we fake it. You'll note that there's no SetPostion on the Somfy MyLink or Tahoma UIs.

**Comment by MattGeale** [comment]
> 

It turns out I was jumping the gun. SetPosition IS working, but I needed to "reset" the blinds from the bond app before set position started working. Its hard to explain, but the tldr is that SetPosition IS working, but I can only call/request them via the Bond app - is there a way to control it from the home app via a slider?

**Comment by chrismerck** [comment]
> is there a way to control it from the home app via a slider?

I don't think that SetPosition is supported in the current `homebridge-bond` plugin. However it might get added in a future version (?).

We're also adding SetPosition to many more shades in the forthcoming V3 firmware release, so it might get more interesting.

**Comment by MattGeale** [comment]
@chrismerck Ah ok. so currently, SetPosition only works/responds if you do it through the bond app with the homebridge app only doing 0 or 100%. All good. Ill create a feature request to see if it can be moved into the Homebridge app in upcoming releases.

**Comment by Fofer** [comment]
> @chrismerck Ah ok. so currently, SetPosition only works/responds if you do it through the bond app with the homebridge app only doing 0 or 100%. All good. Ill create a feature request to see if it can be moved into the Homebridge app in upcoming releases.

Thanks so much for passing along this request. I've been reading along and learning so much along the way.  My Bond bridge is working fine  with my new shades but if Homebridge could somehow support SetPosition too, in order to extend it to HomeKit, that would justify an upgrading to the Bond Pro hub instead.  So I'll continue following this discussion, is there a link so I can follow (or chime in with support for) your feature request?

**Comment by aarons22** [comment]
Thanks for the request. I’m closing this because the plugin cannot reliably implement it with the state/actions currently exposed by Bond and HomeKit. If newer Bond firmware exposes explicit support for this, please open a new feature request with the device actions and state payload.
